compiler/rustc_middle/src/traits/select.rs

```diff
@@ -174,6 +174,10 @@ pub enum SelectionCandidate<'tcx> {
 pub enum EvaluationResult {
     /// Evaluation successful.
     EvaluatedToOk,
+    /// Evaluation successful, but need to rerun because opaque types got
+    /// hidden types assigned without it being known whether the opaque types
+    /// are within their defining scope
+    EvaluatedToOkModuloOpaqueTypes,
```

Comment:
Is `must_apply_modulo_regions()`'s implementation still correct? It looks like that will return `true` for `EvaluatedToOkModuloOpaqueTypes`; is that valid?

Reply:
yea, it's not, but I also wasn't able to create a test that actually was a problem. Just to be safe, I moved the opaque types down